### PR TITLE
Convert XMLHttpRequest to Fetch API

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

This converts the instances of XMLHttpRequest in the code to tthe Fetch API.

The Fetch API is cleaner, faster and more readable than the old XMLHttpRequest.

Surface testing was done, but I might've missed stuff.

The various editors aren't covered in this since there are ES6 PRs for them currently opened.

**Issues**

### CODE REVIEW

File: src/plugins/htmlVideoPlayer/plugin.js [hunk 1]
```diff
@@ -1017,31 +1017,22 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
...
-                    reject(e);
-                    decrementFetchQueue();
-                };
-
-                xhr.send();
-            });
+            try {
+                const response = await fetch(url);
+                decrementFetchQueue();
+                return await response.json;
```
Comment (grafixeyehero) [inline src/plugins/htmlVideoPlayer/plugin.js:1031]: missing bracket

File: src/components/packageManager.js [hunk 2]
```diff
@@ -24,53 +24,43 @@ define(['appSettings', 'pluginManager'], function (appSettings, pluginManager) {
...
-
-            xhr.onerror = onError;
-
-            xhr.send();
-        });
+                var promises = plugins.map(function (pluginUrl) {
+                    return pluginManager.loadPlugin(packageManager.mapPath(pkg, pluginUrl));
+                });
+                Promise.all(promises).then(() => {
+                    return;
```
Comment (thornbill) [inline src/components/packageManager.js:49]: I don’t think these returns do what you think they do. 😉

File: src/components/require/requiretext.js [hunk 3]
```diff
@@ -26,14 +26,10 @@ define(function () {
                 url += 'r=0';
             }
 
-            var xhr = new XMLHttpRequest();
-            xhr.open('GET', url, true);
+            const response = await fetch(url);
```
Comment (thornbill) [inline src/components/require/requiretext.js:29]: Any call to fetch needs to check the response status.

File: src/scripts/globalize.js [hunk 4]
```diff
@@ -164,22 +163,17 @@ define(['userSettings', 'events'], function (userSettings, events) {
...
-
-            xhr.onerror = function () {
-                resolve({});
-            };
-            xhr.send();
-        });
+            const response = await fetch(url);
+            const jsonConntent = await response.json();
+
+            if (response.status < 400) {
```
Comment (thornbill) [inline src/scripts/globalize.js:169]: Need to check this before parsing as json or it can throw an exception.

**Comment by dmitrylyzo** [comment]
Hmm, then we have 2 more in that big ES6 PR.
https://github.com/jellyfin/jellyfin-web/blob/4709e76b5a642c60c92bdf1ea2235a2483b463ff/src/components/libraryoptionseditor/libraryoptionseditor.js#L367-L368
https://github.com/jellyfin/jellyfin-web/blob/4709e76b5a642c60c92bdf1ea2235a2483b463ff/src/components/imageOptionsEditor/imageOptionsEditor.js#L93-L94

**Comment by MrTimscampi** [comment]
This seems to have been mostly taken care of during the move to webpack or unrelated changes, since all I can find anymore is a reference in the Packages and a hack around Fetch not supporting the `file:` protocol.

So I think this is safe to close without really losing anything.

**Pull request closed without merging**